cli/src/main/java/dev/starfix/Starfix.java

```diff
@@ -116,25 +127,64 @@ public static boolean isBlob(String url){
         return Pattern.matches(pattern,url);
     }
 
-    // Function yo determine if the current OS is Windows
+    // Function to determine if the current OS is Windows
     public static boolean isWindows() {
         return System.getProperty("os.name").toLowerCase().contains("windows");
     }
 
+    // Function to determine if the current OS is Linux
+    public static boolean isLinux() {
+        return System.getProperty("os.name").toLowerCase().contains("linux");
+    }
+
+    // Function to determine if the current OS is MacOS
+    public static boolean isMac() {
+        return System.getProperty("os.name").toLowerCase().contains("mac");
+    }
+
+    // Function to fetch config file path
+    public static File getConfigFilePath(){
+        String userHome = System.getProperty("user.home");
+        return new File(userHome + "/.config/starfix.yaml");
+    }
+
     // Function to fetch config file
     public static File getConfigFile() {
-        String userHome = System.getProperty("user.home"); // Get User Home Directory: /home/user_name
-        File configDir = new File(userHome+ "/.config");
-
-        if(!configDir.exists()){ // If .config directory does not exist we create it
-            if(!configDir.mkdirs()){ // If creation failed
-                throw new IllegalStateException("Cannot create .config directory: " + configDir.getAbsolutePath());
+        File configFilePath = getConfigFilePath();
+        
+        if(!configFilePath.getParentFile().exists()){// Check if parent directory exists
+            if(!configFilePath.getParentFile().mkdirs()){// Create parent dir if not exist
+                throw new IllegalStateException("Cannot create .config directory: " + configFilePath.getParentFile().getAbsolutePath());
             }
         }
-        return new File(userHome + "/.config/starfix.yaml");
+        return configFilePath;
 
     }
+    
+    // Function to load default config 
```

Comment:
it does not load anything ,right ? it just setups default configuration.

Reply:
Yeah it just sets up default config.